### PR TITLE
func new will now run init if the proper files have not already been created for the function app.

### PR DESCRIPTION
**Pull request by gzuber**
Cherry Pick of #2028

### CODE REVIEW

**Comment by soninaren** [comment]
Looks like the tests are broken.

**Comment by gzuber** [comment]
@soninaren 

I'm a little confused by this error message, is this really something wrong with specifically my PR?

```
Starting test execution, please wait...

A total of 1 test files matched the specified pattern.
Results File: D:\a\1\s\test\Azure.Functions.Cli.Tests\TestResults\VssAdministrator_fv-az90_2020-07-02_03_11_35.trx

Testhost process exited with error: It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '2.2.0' was not found.
  - The following frameworks were found:
      1.0.16 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      1.1.13 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.1 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.2 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.3 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.4 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.5 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.6 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.8 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.9 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.10 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.11 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.12 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.13 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.14 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.15 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.16 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.17 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.18 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.19 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      3.1.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      3.1.1 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      3.1.2 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      3.1.3 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      3.1.4 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      3.1.5 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
You can resolve the problem by installing the specified framework and/or SDK.
The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=2.2.0&arch=x64&rid=win10-x64
. Please check the diagnostic logs for more information.
Test Run Aborted.
```

Is this an issue with the workers running the test?

**Comment by soninaren** [comment]
This is a devops issue, there is a thread on teams on hiw to fix it. We faced this for host as well.

File: src/Azure.Functions.Cli/Actions/LocalActions/CreateFunctionAction.cs [hunk 1]
```diff
@@ -87,7 +93,15 @@ public async override Task RunAsync()
             }
 
             var workerRuntime = GlobalCoreToolsSettings.CurrentWorkerRuntimeOrNone;
+            if (!FileSystemHelpers.FileExists(Path.Combine(Environment.CurrentDirectory, "local.settings.json")))
```
Comment (soninaren) [inline src/Azure.Functions.Cli/Actions/LocalActions/CreateFunctionAction.cs:96]: might be worth checking for `host.json` as well

**Review by soninaren — APPROVED** [review]
Left a nit comment. Looks good otherwise.

**Review by ankitkumarr — APPROVED** [review]

**Pull request merged**